src/braket/experimental/autoqasm/doc/decorators.md

```diff
@@ -8,26 +8,26 @@ There are a handful of decorators available through AutoQASM. Each one attaches
 
 This decorator marks the entry point to a quantum program.
 
-You can include gates, pulse control, classical control and subroutine calls. When you call the function wrapped by `@aq.main`, you will get a `Program` object. The `Program` object can be executed on [devices available through Amazon Braket](https://docs.aws.amazon.com/braket/latest/developerguide/braket-devices.html), including local simulators. The code snippet below creates a quantum program with `@aq.main` and runs it on the `Device` instantiated as `device`.
+You can include gates, pulse control, classical control and subroutine calls. The function wrapped by `@aq.main` is converted into a `Program` object. The `Program` object can be executed on [devices available through Amazon Braket](https://docs.aws.amazon.com/braket/latest/developerguide/braket-devices.html), including local simulators. The code snippet below creates a quantum program with `@aq.main` and runs it on the `Device` instantiated as `device`.
 
 ```
-@aq.main(num_qubits=5)
-def ghz_state(max_qubits):
-    """Create a GHZ state from a variable number of qubits."""
+num_qubits = 5
+
+@aq.main(num_qubits=num_qubits)
+def ghz_state():
+    """Create a GHZ state of size num_qubits."""
     h(0)
-    for i in aq.range(1, max_qubits):
+    for i in aq.range(1, num_qubits):
         cnot(0, i)
-    measure(list(range(max_qubits))) 
-    
-ghz_state_program = ghz_state(max_qubits=5)
+    measure(list(range(num_qubits))) 
```

Comment:
Is the list cast really necessary? I would hope that measure can take in any iterable

Reply:
This is no longer necessary. `measure(range(num_qubits))` would works as well.